pkg/accounting/tally/tally.go

```diff
@@ -101,15 +101,46 @@ func (t *Tally) calculateAtRestData(ctx context.Context) (latestTally time.Time,
 	}
 	nodeData = make(map[storj.NodeID]float64)
 
+	var currentBucket string
+	var bucketCount int64
+	var totalStats, currentBucketStats Stats
+
 	err = t.pointerdb.Iterate("", "", true, false,
 		func(it storage.Iterator) error {
 			var item storage.ListItem
 			for it.Next(&item) {
+
 				pointer := &pb.Pointer{}
 				err = proto.Unmarshal(item.Value, pointer)
 				if err != nil {
 					return Error.Wrap(err)
 				}
+
+				pathElements := storj.SplitPath(storj.Path(item.Key))
+				// check to make sure there are at least *4* path elements. the first three
+				// are project, segment, and bucket name, but we want to make sure we're talking
+				// about an actual object, and that there's an object name specified
+				if len(pathElements) >= 4 {
+					project, segment, bucketName := pathElements[0], pathElements[1], pathElements[2]
+					bucketID := storj.JoinPaths(project, bucketName)
+
+					// paths are iterated in order, so everything in a bucket is
+					// iterated together. When a project or bucket changes,
+					// the previous bucket is completely finished.
+					if currentBucket != bucketID {
+						bucketCount++
+						if currentBucket != "" {
+							// report the previous bucket and add to the totals
+							currentBucketStats.Report("bucket.")
```

Comment:
Do we want to submit each bucket as the same metric or separately?

Reply:
Per my understanding we are currently not submitting any info based on bucket. we are aggregating the needed numbers (total segments, total files, total size of the files.... etc) while parsing thru pointerdb. Did this clarify the comment?

Reply:
oh definitely the same. basically, metric names should all be fixed by compile time. if metric names are dynamic like that the time series db will get overrun

Reply:
Is there a way to make monkit aware of the multiple buckets per call? So that it would collect summaries rather than the values? Currently it seems to be aggregating as `0:alpha, 0:beta, 0:gamma, 1:alpha, 1:beta, 1:gamma` rather than `0:[alpha, beta, gamma] 1:[alpha, beta, gamma]`. Not sure how much it makes a difference though.